system/mp/mpcomm.cpp

```diff
@@ -2178,194 +2265,272 @@ void CMPConnectThread::startPort(unsigned short port)
     if (!listen)
         return;
     running = true;
+    if (acceptThreadPoolSize)
+    {
+        class CFactory : public CInterfaceOf<IThreadFactory>
+        {
+            CMPConnectThread &owner;
+        public:
+            CFactory(CMPConnectThread &_owner) : owner(_owner)
+            {
+            }
+        // IThreadFactory
+            IPooledThread *createNew() override
+            {
+                class CMPConnectionThread : public CInterfaceOf<IPooledThread>
+                {
+                    CMPConnectThread &owner;
+                    Owned<ISocket> sock;
+                public:
+                    CMPConnectionThread(CMPConnectThread &_owner) : owner(_owner)
+                    {
+                    }
+                // IPooledThread
+                    virtual void init(void *param) override
+                    {
+                        sock.set((ISocket *)param);
+                    }
+                    virtual void threadmain() override
+                    {
+                        constexpr unsigned timeoutMs = 5000;
+                        Owned<ISocket> handledSock = sock.getClear();
+                        if (owner.handleAcceptedSocket(handledSock.getLink(), timeoutMs, false))
```

Comment:
Not worth changing, but it is counter-intuitive that a function called handleAcceptedSocket() returns true if it fails to handle it.

Reply:
agree, and pondered this, but really true|false is insufficient. As it stands returning false means it did not timeout, but it doesn't mean it succeeded. Various failure conditions also return false.
So really it should return a state, but I only care about 1 of them (timeout).